### DISCUSSION TITLE
gatsby-plugin-google-tagmanager and defaultDataLayer to sent 'consent' data

### DISCUSSION THREAD

**Discussion by kuworking** [gatsbyjs/gatsby#28752] [Help]
Linked to https://github.com/gatsbyjs/gatsby/issues/20963 where the answer said that this was not possible

I am wondering whether this is different with the `consent` tag https://developers.google.com/gtagjs/devguide/consent 

As far as I see it, wouldn't suffice to state the `defaultDataLayer` to send the equivalent to this?

```js
          window.dataLayer.push('consent', 'default', {
            ad_storage: 'denied',
            analytics_storage: 'denied',
            wait_for_update: 500,
          })
```

Which brings me the question, how should this be done? My guess is removing `'consent'` and `'default'` (?)

```js
    {
      resolve: `gatsby-plugin-google-tagmanager`,
      options: {
        id: gtm,
        includeInDevelopment: false,
        // datalayer to be set before GTM is loaded
        // should be a stringified object or object
        //
        // Defaults to null
        defaultDataLayer: {
          ad_storage: 'denied',
          analytics_storage: 'denied',
          wait_for_update: 500,
        },
      },
    },
```

**Comment by BenWildeman** [comment]
@kuworking can this be done?

**Reply by Misyuk-T** [reply]
@kuworking faced with the same issue. Any updates?

**Comment by azmaat1** [comment]
yas
